legacy/src/main/java/org/opensearch/sql/legacy/pit/PointInTimeHandlerImpl.java

```diff
@@ -37,47 +40,55 @@ public PointInTimeHandlerImpl(Client client, String[] indices) {
     this.indices = indices;
   }
 
-  /** Create PIT for given indices */
+  /**
+   * Constructor for class
+   *
+   * @param client OpenSearch client
+   * @param pitId Point In Time ID
+   */
+  public PointInTimeHandlerImpl(Client client, String pitId) {
+    this.client = client;
+    this.pitId = pitId;
+  }
+
+  /**
+   * Create PIT for given indices
+   *
+   * @return Point In Time creation status
+   */
   @Override
-  public void create() {
+  public boolean create() {
     CreatePitRequest createPitRequest =
         new CreatePitRequest(
             LocalClusterState.state().getSettingValue(SQL_CURSOR_KEEP_ALIVE), false, indices);
-    client.createPit(
-        createPitRequest,
-        new ActionListener<>() {
-          @Override
-          public void onResponse(CreatePitResponse createPitResponse) {
-            pitId = createPitResponse.getId();
-            LOG.info("Created Point In Time {} successfully.", pitId);
-          }
-
-          @Override
-          public void onFailure(Exception e) {
-            LOG.error("Error occurred while creating PIT", e);
-          }
-        });
+    ActionFuture<CreatePitResponse> execute =
+        client.execute(CreatePitAction.INSTANCE, createPitRequest);
+    try {
+      CreatePitResponse pitResponse = execute.get();
+      pitId = pitResponse.getId();
+      LOG.info("Created Point In Time {} successfully.", pitId);
+      return true;
```

Comment:
What is this boolean used for?

Where is false scenario?

Reply:
Currently I see this boolean is used only in unit tests for assertions. When PIT creation fails, I'm throwing exception instead of returning false to stop the query execution.

Reply:
we don't change interface methods just for unit tests.
In unit tests, use some other way by checking no exceptions are made and change this method to void.